src/browser/wisp_network.js

```diff
@@ -0,0 +1,320 @@
+"use strict";
+const DEFAULT_DOH_SERVER = "cloudflare-dns.com";
+
+/**
+ * @constructor
+ *
+ * @param {BusConnector} bus
+ * @param {*=} config
+ */
+function WispNetworkAdapter(wisp_url, bus, config)
+{
+
+    this.register_ws(wisp_url);
+    config = config || {};
+    this.bus = bus;
+    this.id = config.id || 0;
+    this.router_mac = new Uint8Array((config.router_mac || "52:54:0:1:2:3").split(":").map(function(x) { return parseInt(x, 16); }));
+    this.router_ip = new Uint8Array((config.router_ip || "192.168.86.1").split(".").map(function(x) { return parseInt(x, 10); }));
+    this.vm_ip = new Uint8Array((config.vm_ip || "192.168.86.100").split(".").map(function(x) { return parseInt(x, 10); }));
+    this.masquerade = config.masquerade === undefined || !!config.masquerade;
+    this.vm_mac = new Uint8Array(6);
+    this.doh_server =  config.doh_server || DEFAULT_DOH_SERVER;
+    this.tcp_conn = {};
+
+    this.bus.register("net" + this.id + "-mac", function(mac) {
+        this.vm_mac = new Uint8Array(mac.split(":").map(function(x) { return parseInt(x, 16); }));
+    }, this);
+    this.bus.register("net" + this.id + "-send", function(data)
+    {
+        this.send(data);
+    }, this);
+}
+
+WispNetworkAdapter.prototype.register_ws = function (wisp_url) {
+    this.wispws = new WebSocket(wisp_url.replace("wisp://", "ws://").replace("wisps://", "wss://"));
+    this.wispws.binaryType = "arraybuffer";
+    this.wispws.onmessage = (event) => {
+        this.process_incoming_wisp_frame(new Uint8Array(event.data));
+    };
+    this.wispws.onclose = () => {
+        this.register_ws(wisp_url);
```

Comment:
This should probably have a `setTimeout` of 10 or so seconds, just so we don't hammer the server if there are problems with the connection.